src/tools/pkcs11-tool.c

```diff
@@ -2863,11 +2863,12 @@ static int gen_keypair(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 				if (!find_mechanism(slot, CKF_GENERATE_KEY_PAIR, mtypes, mtypes_num, &opt_mechanism))
 					util_fatal("Generate RSA mechanism not supported");
 
-			if (size == NULL)
-				util_fatal("Unknown key pair type %s, expecting RSA:<nbytes>", type);
 			key_length = (unsigned long)atol(size);
```

Comment:
Does the atol handle NULL parameters correctly? I did not find this confirmed in the manual pages.

Reply:
My bad. You are right.